util/topgen/lib.py

```diff
@@ -21,6 +21,47 @@
 from .intermodule import get_dangling_im_def # noqa : F401 # isort:skip
 
 
+class Name(object):
```

Comment:
This class is actually something I have moved over from `c.py`.
I've put it into `lib.py` to share it more broadly and avoid circular imports.

Since this is used in many places, I suggest to leave it at `Name` for now.
But I cleaned up the other things you have mentioned.